### PR TITLE
Subscribers Page: Added export as csv to Subscribers Page

### LINKED ISSUE #78364: Subscribers Page: Add ability to download email subscribers as CSV.
Add ability to download email subscribers when clicked on the "triple-dot" dropdown menu.

Follow the logic as on the existing implementation on the old Subscribers Page, it should be just a simple API call.

image

image

### PR DESCRIPTION
**Pull request by JuanLucha**
Closes https://github.com/Automattic/wp-calypso/issues/78364

## Proposed Changes

This PR adds the "export to CSV" feature to the Subscribers Page. This will export a CSV file with the emails of the email subscribers.

**Note:** Only the email subscribers will appear in the CSV, so you can find more people in the list on the Subscribers Page than in the CSV file.

## Testing Instructions

1. Apply this PR and start the application.
2. Go to `http://calypso.localhost:3000/subscribers/{the domain of your site}`.
3. Click on the ellipsis menu in the top right corner of the screen and then on the `Download email subscribers as CSV` link.
Screenshot 2023-06-20 at 18 20 07 (1)
4. It should download a file with the name like `[domain-of-the-site]-[id-of-the-site]-email`. If you open it, you will see the list of emails of the email subscribers on the Subscribers Page.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/subscribers/components/subscribers-header-popover/subscribers-header-popover.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			<PopoverMenu
+				position="bottom left"
+				onClose={ () => setIsVisible( false ) }
+				isVisible={ isVisible }
+				context={ buttonRef.current }
+				className="subscriber-popover"
+				focusOnShow={ false }
+			>
+				<PopoverMenuItem href={ downloadCsvLink } onClick={ onDownloadCsvClick }>
+					{ translate( 'Download email subscribers as CSV' ) }
```
Comment (phcp) [inline client/my-sites/subscribers/components/subscribers-header-popover/subscribers-header-popover.tsx:56]: The design on Figma shows `Download CSV` and the implementation has `Download email subscribers as CSV`. Is there an agreement about this button label (considering the note about the CSV only with email subscribers)?

**Review by phcp — APPROVED** [review]
Great job! The PR is working as expected and the code looks good, I liked the way you did to share the styles between popovers, super clear approach. Just left one question regarding the button label, but feel free to move forward and deploy.

File: client/my-sites/subscribers/components/subscribers-header-popover/subscribers-header-popover.tsx (hunk 1 above)
Comment (JuanLucha) [reply client/my-sites/subscribers/components/subscribers-header-popover/subscribers-header-popover.tsx:56]: You are right. But, honestly, "Download CSV" makes the impression that all the users in the list will be downloaded when the only ones being downloaded are the email subscribers. I think `Download email subscribers as CSV` is more accurate on what the action will do.

Comment (JuanLucha) [reply client/my-sites/subscribers/components/subscribers-header-popover/subscribers-header-popover.tsx:56]: For now we leave the copy like this (discussed here: p1687337763769789-slack-C02TCEHP3HA )

**Pull request merged**